### ISSUE TITLE
Bug: Warning: unknown raw line 'mksock'

### ISSUE THREAD

**Issue by mbideau** [mbideau/btrfs-diff-sh#3]
I was trying to benc both versions on the snapshots of my root ssd, but I encountered a couple of fatal errors:
```
4,9G	root_snapshot.20210816/
6,7G	root_snapshot.20210905/
```
```
[root@home-server /mountpoints/root/snapshots/root]# time sudo btrfs-diff root_snapshot.20210816/ root_snapshot.20210905/
Warning: unknown raw line 'mksock          ./root_snapshot.20210905/o59141-618-0'
Fatal error: when renaming './root_snapshot.20210905/o59141-618-0' to './root_snapshot.20210905/etc/samba/private/msg.sock/6555', the source wasn't found in the objects buffer

real	12m40,858s
user	13m0,688s
sys	4m48,570s
```

_Originally posted by @Mek101 in https://github.com/mbideau/btrfs-diff-go/issues/2#issuecomment-913127997_

**Comment by mbideau** [comment]
Fixed by commit f53c427

**Comment by mbideau** [comment]
Hey @Mek101,
Could you try again, now that the bug is fixed ?

This one is simple and slow but deliver good results.
The Go version still have a bug, and may be more. I will compare it against the Shell version that I consider the reference.

**Comment by Mek101** [comment]
Unfortunately it seems that the process ends up stalling, after about 50~ minutes it didn't produce any output. Same command line of the opening comment

**Comment by Mek101** [comment]
I tested it again on other snapshots with fewer changes: it worked well with them with no sign of bugs. It seems that it was just extremely slow

**Comment by mbideau** [comment]
Hey @Mek101,
Sorry for being late, I was soo busy those last days ...
Thank you again for your feedback.
Yes, it is a shame that the shell version is soo slow, but it was expected (not that slow though) since it does a crazy amount of file I/O the way I have implemented it. It was a known caveat of the design since the beginning. I wanted something that were working for me as quick as possible, so I sacrifice a little bit (actually a huge part) of the implementation to get results fast.
But that slowness had got me the motivation to fix the Go version, which is super fast, and now almost bug free, thanks to you (when I would have fixed your last found bug :sweat_smile: ).
BTW if you ever come to Marseille (France), I owned you a beer (or a bunch of it) :wink:
